### ISSUE TITLE
Arm64 machines cannot build arm64 native executables

### ISSUE THREAD

**Issue by kevinxufei** [xamarin/xamarin-macios#16152]
**Affected Branch:** release-17.3
**Affected Build:** 17.3.6 build 20

**Configuration:**
> - .NET SDK version: 6.0.401
> - MAUI: 6.0.540
> - Xcode: 14.0 (14A309)
> - Xamarin.iOS: 16.0.0.72
> - Xamarin.Mac: 8.12.0.2

**Steps to reproduce:**

> 1. Launch VSMac and create a MacOS Application(Other->Custom) project
> 2. Click "Run" button

**Actual result:**  
Debug MacOS Application failed with "Unable to attach to CoreCLR. Unknown Error: 0x80131c3c"
![image](https://user-images.githubusercontent.com/86938401/192503105-67993fec-1906-43f5-a3cc-248a30f43b0e.png)

![image](https://user-images.githubusercontent.com/86938401/192503167-b5a1a678-9d85-4fd3-85dd-75cfaf204f41.png)

**Comment by rolfbjarne** [comment]
You have to change the RuntimeIdentifier in the project file:

```xml
<PropertyGroup>
    <RuntimeIdentifier>osx-arm64</RuntimeIdentifier>
</PropertyGroup>
```

Admittedly we could probably have a better default value here.

**Comment by mrward** [comment]
This bug is fixed in VS Mac 17.4, since VS Mac 17.4 will determine the architecture of the executable instead of assuming it is arm64. On the original bug filed against VS Mac I suggested that a bug is filed here for the Microsoft.macOS sdk to see if it should be using x64 by default when on an arm64.

**Comment by rolfbjarne** [comment]
Fixed in https://github.com/xamarin/xamarin-macios/pull/18495.
